ICSharpCode.Decompiler/TypeSystem/MetadataModule.cs

```diff
@@ -324,6 +324,14 @@ public IModule ResolveModule(AssemblyReferenceHandle handle)
 			return LazyInit.GetOrSet(ref referencedAssemblies[row], module);
 		}
 
+		int GetRowNumber(AssemblyReferenceHandle handle)
+		{
+			var row = MetadataTokens.GetRowNumber(handle);
+			if (row == -1 )
+				row = MetadataTokens.GetRowNumber(metadata, handle);
```

Comment:
Wouldn't it be faster to always use the `MetadataReader` overload?

Reply:
> Wouldn't it be faster to always use the MetadataReader overload?

It would, this is what I was talking about here :) : 
> * In theory I think we could only use `MetadataTokens.GetRowNumber(metadata, handle);` instead of that being the backup case but I wasn't 100% confident on that so I've added it as an edge case for now.

Reply:
If you're more confident than I am that using the overload won't cause problems I'll refactor and remove the new method.

Reply:
I don't see why this would cause problems.